cmd/utils/flags.go

```diff
@@ -1738,37 +1738,36 @@ func SetEthConfig(ctx *cli.Context, stack *node.Node, cfg *ethconfig.Config) {
 		cfg.NetworkId = *cfg.Genesis.GetNetworkID()
 	}
 
-	if ctx.GlobalIsSet(EthProtocolsFlag.Name) {
-		protocolVersions := SplitAndTrim(ctx.GlobalString(EthProtocolsFlag.Name))
-		if len(protocolVersions) == 0 {
-			Fatalf("--%s must be comma separated list of %s", EthProtocolsFlag.Name, strings.Join(strings.Fields(fmt.Sprint(eth.DefaultProtocolVersions)), ","))
-		}
+	// Set the supported ETH Protocol Versions
+	protocolVersions := SplitAndTrim(ctx.GlobalString(EthProtocolsFlag.Name))
+	if len(protocolVersions) == 0 {
+		Fatalf("--%s must be comma separated list of %s", EthProtocolsFlag.Name, strings.Join(strings.Fields(fmt.Sprint(eth.DefaultProtocolVersions)), ","))
+	}
 
-		seenVersions := map[uint]interface{}{}
-		for _, versionString := range protocolVersions {
-			version, err := strconv.ParseUint(versionString, 10, 0)
-			if err != nil {
-				Fatalf("--%s has invalid value \"%v\" with error: %v", EthProtocolsFlag.Name, versionString, err)
-			}
+	seenVersions := map[uint]interface{}{}
+	for _, versionString := range protocolVersions {
+		version, err := strconv.ParseUint(versionString, 10, 0)
+		if err != nil {
+			Fatalf("--%s has invalid value \"%v\" with error: %v", EthProtocolsFlag.Name, versionString, err)
+		}
 
-			if _, duplicate := seenVersions[uint(version)]; duplicate {
-				Fatalf("--%s has duplicate version of %v", EthProtocolsFlag.Name, versionString)
-			}
+		if _, duplicate := seenVersions[uint(version)]; duplicate {
+			Fatalf("--%s has duplicate version of %v", EthProtocolsFlag.Name, versionString)
+		}
 
-			isValid := false
-			for _, proto := range eth.DefaultProtocolVersions {
-				if proto == uint(version) {
-					isValid = true
-					seenVersions[uint(version)] = nil
-					break
-				}
+		isValid := false
+		for _, proto := range eth.DefaultProtocolVersions {
```

Comment:
This validates the passed protocol version using `DefaultProtocolVersions` as the list of valid options (rather than as -- one might assume -- a list of _defaults_). Is this desired? Is there a list of eligibile-but-not-necessarily-default versions?

Reply:
Nice catch. Indeed the `eth.DefaultProtocolVersions` is misleading here. 
The intention was, that this value holds all the supported networks, the same way the used to have in the protocols.go (go-ethereum, till recently), and is being used as the Default value for the flag. 
So we have to either rename it or use a second var for it. 
I am on this one...

Reply:
Resolved by 15e0f16